deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/recurrent/LayerNormalizationWholeMinibatch.java

```diff
@@ -0,0 +1,21 @@
+package org.deeplearning4j.nn.layers.recurrent;
+
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+
+/**
+ * Created by fac2003 on 4/6/17.
+ */
+public class LayerNormalizationWholeMinibatch implements LayerNormalization {
+    @Override
+    public void normalize(INDArray activations) {
+        // normalize across entires mini-batch:
+        Number stdev = activations.stdNumber();
+        activations.subi(Nd4j.mean(activations));
+        final double stdNumber = stdev.doubleValue();
+        if (!Double.isNaN(stdNumber)) {
```

Comment:
I prefer to focus on correctness and worry about optimization later if shown to matter by profiling in various scenarios. 
In this case, also, if you look at the figure, the efficiency of training is about doubled (need half the number of epochs to get to the same score), so I am not too concerned by slightly inefficient code.
If you want to optimize, I would suggest to add a hasAnyNaN method to INDArray. It should return true if any element (i.e., first one encountered) is NaN. This can be parallelized any way you like. I can create an issue if this is something you would find useful (I was surprised not to find this method).